### ISSUE TITLE
Mouseover macro conditional behaviour changed since Wrath 3.4.1

### ISSUE THREAD

**Issue by lordblackadder** [Stanzilla/WoWUIBugs#382]
Since the new patch, `@mouseover` macro conditionals behave differently when the range with your mouseover is over ~50y. With a macro such as `/cast [@mouseover, help] Holy Light` the expected behaviour is that it will cast on your mouseover if it's a friendly unit, and otherwise nothing will happen.

Since the new patch, if you are beyond ~50y the spell be cast on your target (or self). This happens regardless of whether the spell is in range. For instance, it occurs with Beacon of Light even though the spell has 60y range.

Easy way to reproduce with a holy paladin:

1. Put Beacon of Light (not macroed) on your action bar slot 1, and a BoL macro `/cast [@mouseover, help] Beacon of Light` on slot 2
2. Stand at ~30y from a party member and target a non-party unit closer than that (let's say 10y from the paladin)
3. Start casting the BoL macro on the party member while moving away from that unit
4. At around ~50y the macro will fail and cursor will change to a glowy glove, as if the target is out of range or it attempts to cast on your target (because BoL cannot be cast at units outside your group)
5. Press Escape to cancel your glowly glove cursor
6. Target your party member and press action bar slot 1 (the non-macroed BoL) and it will succeed

To summarise: A spell that is within its cast range will succeed when using a normal cast (such as dragging from the spellbook) but fail to cast if that same target unit is selected through `@mouseover`. This also occurs with spells having shorter range: if a spell has 40y range and you attempted to use a similar macro while out of range, you would previously get the glowy glove cursor. Now it will instead go to your target or self even if the macro doesn't have `[]` or `[@target]` conditionals.

The bug is also described on EU wow forums, including video proof:
https://eu.forums.blizzard.com/en/wow/t/mouseover-macros-are-bugged-after-patch-341-out-of-range-weird-bug/423877
